packages/shared/src/lib/query.ts

```diff
@@ -299,3 +300,17 @@ export const updateReadingHistoryListPost = ({
     queryClient.setQueryData(queryKey, oldData);
   };
 };
+
+export function flattenInfiniteQuery<
```

Comment:
More generic, do we need a infiniteQuery then if we expect to flatten the whole result anyway?
EG wouldn't we just one a once-off "all categories"?

Reply:
Oh, right. I haven't actually have thought of that 😂

Reply:
Though still a useful utility as often flatten results. However, in this case, we don't have to if we just use `useQuery`.